### DISCUSSION TITLE
Identifying issues and pull requests from the community

### DISCUSSION THREAD

**Discussion by pdurbin** [opendp/opendp#354] [Getting Involved] [answered]
I just noticed another pull request come in from the community (🎉) and it reminded me that I'd like to share with everyone a trick I use to identify issues and pull requests from non-core contributors.

At a high level, here are the steps:

- Gather a list of GitHub usernames for core contributors.
- Add `-author:userNameHere` for each core contributor to your search.
- Optionally, specify if you'd like to see only issues, only pull requests, only open, only closed, etc.

Obviously, the list of core contributors will change over time so queries have to be adjusted but right now I'd say the core contributors to the OpenDP library repo (people who come to standup, etc.) are Andy, Mike, Raman, Ethan, and me. So in the queries below, you'll see `-author:andrewvyrros -author:Shoeboxam -author:raprasad -author:ecowan -author:pdurbin`.

## Issues

Now, we can go to any GitHub page and enter the following search for issues:

`is:issue repo:opendp/opendp -author:andrewvyrros -author:Shoeboxam -author:raprasad -author:ecowan -author:pdurbin`

Here's a direct link and a screenshot: https://github.com/search?q=is%3Aissue+repo%3Aopendp%2Fopendp+-author%3Aandrewvyrros+-author%3AShoeboxam+-author%3Araprasad+-author%3Aecowan+-author%3Apdurbin

![Screen Shot 2021-10-01 at 10 28 15 AM](https://user-images.githubusercontent.com/21006/135638436-45fbba99-62a6-4421-86ed-353af8cdf0fb.png)

## Pull requests

Here's a search for pull requests:

`is:pr repo:opendp/opendp -author:andrewvyrros -author:Shoeboxam -author:raprasad -author:ecowan -author:pdurbin`

Here's a direct link and a screenshot: https://github.com/search?q=is%3Apr+repo%3Aopendp%2Fopendp+-author%3Aandrewvyrros+-author%3AShoeboxam+-author%3Araprasad+-author%3Aecowan+-author%3Apdurbin

![Screen Shot 2021-10-01 at 10 28 29 AM](https://user-images.githubusercontent.com/21006/135638439-27b6f0ac-2cb9-446d-b9f9-83b69e0d8979.png)

With this information, we can answer questions like this:

- How many external contributors do we have?
- What issues has the community opened?
- What pull requests has the community opened?
- Who in the community should we be thanking for their contribution?

I welcome any thoughts or discussion on this! It might be nice to someday automate this somehow (a script? a bookmarklet?) but I haven't gotten around to it. It seems like a trick that many open source projects could use, since so many are on GitHub.

**Comment by Shoeboxam** [comment, marked as answer]
Marking this "question" as answered. Thanks for the pointer Phil!
